lib/domain/GoogleFunctionProvider.js

```diff
@@ -0,0 +1,127 @@
+const config = require("../support/config");
+const fetch = require('node-fetch');
+const { GoogleAuth } = require("google-auth-library");
+const { FunctionServiceClient } = require('@google-cloud/functions').v2;
+// const { request } = require('gaxios');
+
+const log = require("../support/log");
+
+class GoogleFunctionProvider {
+  constructor() {
+  }
+
+  async authenticate(url){
+    try{
+      const sa_auth = JSON.parse(Buffer.from(config.googleCloudCredentials, "base64").toString('ascii'))
+      const client = new GoogleAuth().fromJSON(sa_auth);
+      const token = await client.fetchIdToken(url)
+      return `Bearer ${token}`
+    }catch(err){
+      throw err
+    }
+  }
+
+  async run(req) {
+    try {
+      const zoneId = "us-central1"
+      const projectId = "gglobo-bs-events-hdg-dev"
```

Comment:
Ooooops!

IMHO, Keep your private information private

Reply:
lol, not really dangerous, but i changed to env vars, we are making decisions about these ids will be defined by env or function definition endpoint